runtime/gc_ctrl.c

```diff
@@ -550,60 +552,77 @@ static void test_and_compact (void)
 
 CAMLprim value caml_gc_major(value v)
 {
-  CAML_INSTR_SETUP (tmr, "");
+  value exn;
+
+  CAML_EV_BEGIN(EV_EXPLICIT_GC_MAJOR);
   CAMLassert (v == Val_unit);
   caml_gc_message (0x1, "Major GC cycle requested\n");
   caml_empty_minor_heap ();
   caml_finish_major_cycle ();
   test_and_compact ();
   // call finalisers
-  caml_process_pending_actions();
-  CAML_INSTR_TIME (tmr, "explicit/gc_major");
+  exn = caml_process_pending_actions_exn();
+  CAML_EV_END(EV_EXPLICIT_GC_MAJOR);
+  caml_raise_if_exception(exn);
   return Val_unit;
 }
 
 CAMLprim value caml_gc_full_major(value v)
 {
-  CAML_INSTR_SETUP (tmr, "");
+  value exn;
+
+  CAML_EV_BEGIN(EV_EXPLICIT_GC_FULL_MAJOR);
   CAMLassert (v == Val_unit);
   caml_gc_message (0x1, "Full major GC cycle requested\n");
   caml_empty_minor_heap ();
   caml_finish_major_cycle ();
   // call finalisers
-  caml_process_pending_actions();
+  exn = caml_process_pending_actions_exn();
+  if (Is_exception_result(exn)) goto cleanup;
   caml_empty_minor_heap ();
   caml_finish_major_cycle ();
   test_and_compact ();
   // call finalisers
   caml_process_pending_actions();
```

Comment:
```suggestion
```
Either GitHub's display is going wrong, or you forgot to remove this line, right?

Reply:
It was indeed forgotten, sorry about this.